src/main/kotlin/com/example/delivery/service/UserService.kt

```diff
@@ -26,6 +29,20 @@ class UserService(
             ?: throw NotFoundException("User with id $id doesn't exists")
     }
 
+    fun update(id: String, updateUserDTO: UpdateUserDTO): DomainUser {
+        return userRepository.update(id, createUpdateObject(updateUserDTO))?.toDomain()
+            ?: throw NotFoundException("User with id $id doesn't exists")
+    }
+
+    private fun createUpdateObject(updateUserDTO: UpdateUserDTO): Update {
+        val update = Update()
+        with(updateUserDTO) {
+            fullName?.let { update.set(MongoUser::fullName.name, fullName) }
+            phone?.let { update.set(MongoUser::phone.name, phone) }
+        }
+        return update
+    }
+
     fun findAllOrders(id: String): List<DomainOrder> {
```

Comment:
findAllOrders should be renamed to getAllOrders, because this method throws exception.